File: 240429/문자에 따른 명령 2/text-based-commands2.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        String str = sc.next();
        int length = str.length();
        char[] arr = str.toCharArray();
        int dirnum = 0;
        int x = 0;
        int y = 0;
        int[] dx = new int[]{0,1,0,-1};
        int[] dy = new int[]{1,0,-1,0};
        for(int i=0;i<length;i++) {
            if(arr[i]=='L')
                dirnum += 3;
            if(arr[i]=='R')
                dirnum += 1;
            if(arr[i]=='F')
                x += dx[dirnum%4];
                y += dy[dirnum%4];
        }

        System.out.println(x+ " "+y);
    }
}

File: 240424/문자에 따른 명령 2/text-based-commands2.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)
    var dirNum = 0

    var x = 0
    var y = 0

    val dx = intArrayOf(0, 1, 0, -1)
    val dy = intArrayOf(1, 0, -1, 0)

    val commands = sc.next()

    for (i in 0 until commands.length) {
        when (commands[i]) {
            'L' -> { dirNum = (dirNum - 1 + 4) % 4 }
            'R' -> { dirNum = (dirNum + 1) % 4 }
            else -> { 
                x += dx[dirNum] 
                y += dy[dirNum]
            }
        }
    }

    print("$x $y")
}